chase/frontend/app/login/participant/page.tsx

```diff
@@ -1,24 +1,21 @@
 "use client";
-import React, { useState, useContext } from "react";
+import React, { useState, Suspense } from "react";
 import Image from "next/image";
 
 import UsernameLogin from "../../../components/login/username";
 import PasswordLogin from "../../../components/login/password";
+import Loading from "@/app/loading";
+import Link from "next/link";
 
 export default function loginVorsitz() {
   const [loginStage, changeLoginState] = useState(0);
 
   return (
     <>
-      <div
-        className="flex justify-stretch h-screen"
-        style={{ backgroundColor: "#eee" }}
-      >
+      <Suspense fallback={<Loading />} />
```

Comment:
Isn't this the right way to enable the loading animation before the page finishes loading?
https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming

Reply:
I don't know – i though if it applied to page sections, it might as well apply to the whole page. But there might be a simpler way?

Reply:
As far as I know, Next.js does this automatically. Whenever a page render is delayed, e.g. because we need to make an API call first, the `loading.tsx` is shown.

The `<Suspense>` tag can be used to wrap content which can be delayed inside a component. There are overlaps in usage of these two features, but generally saying you use `<Suspense>` to show a part of a component which can be delayed and the `loading.tsx` for a whole page. But to use `<Suspense>` you need to wrap the delayed content inside of the tag like so:

```
<Suspense fallback={<MyCoolLoadingSpinner />}>
  <MyComponentWhichTakesTimeToLoad />
<Suspense />
```